### PR TITLE
static-list-of-posts-js

### PR DESCRIPTION
**Pull request by Q1W2E3R4T5Y6U7I8a**
[DEMO LINK](https://Q1W2E3R4T5Y6U7I8a.github.io/react_static-list-of-posts-js/)

### CODE REVIEW

File: src/components/PostInfo/PostInfo.jsx [hunk 1]
```diff
@@ -1,3 +1,24 @@
...
+
+      <p>
+        {' Posted by '}
+        {post.user && <UserInfo user={post.user} />}
+      </p>
+    </div>
+
+    <p className="PostInfo__body">{post.body}</p>
+
+    {post.comments && post.comments.length > 0 ? (
```
Comment (etojeDenys) [inline src/components/PostInfo/PostInfo.jsx:18]:
```suggestion
    {post.comments?.length > 0 ? (
```

File: src/App.jsx [hunk 2]
```diff
@@ -1,107 +1,39 @@
...
+        },
+        {
+          id: 2,
+          name: 'odio adipisci rerum aut animi',
+          email: 'Nikita@garfield.biz',
+          body: 'quia molestiae reprehenderit quasi aspernatur aut expedita...',
+        },
+      ],
+    },
+  ];
```
Comment (etojeDenys) [inline src/App.jsx:31]: you should import these posts, and add users, and comments there

**Review by etojeDenys — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 2 above)
Comment (Q1W2E3R4T5Y6U7I8a) [reply src/App.jsx:31]: how? I don't quite understand.

Comment (Viktor-Kost) [reply src/App.jsx:31]: here were an example at the forked repo
```
// import postsFromServer from './api/posts.json';
// import commentsFromServer from './api/comments.json';
// import usersFromServer from './api/users.json';
```

Comment (Viktor-Kost) [reply src/App.jsx:31]: so we will see not only one user
Screenshot 2023-12-09 at 23 07 57

File: src/components/CommentInfo/CommentInfo.jsx [hunk 3]
```diff
@@ -1,3 +1,12 @@
-export const CommentInfo = () => (
-  <>Put the comment here</>
+export const CommentInfo = ({ name, email, body }) => (
+  <div className="CommentInfo">
+    <div className="CommentInfo__title">
+      <strong className="CommentInfo__name">{name}</strong>
+      {' by '}
```
Comment (Viktor-Kost) [inline src/components/CommentInfo/CommentInfo.jsx:5]: it's better to use `span`
```suggestion
      <span> by </span>
```

File: src/components/PostInfo/PostInfo.jsx [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+import './PostInfo.scss';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
+
+export const PostInfo = ({ title, user, body, comments }) => (
+  <div className="PostInfo">
+    <div className="PostInfo__header">
+      <h3 className="PostInfo__title">{title}</h3>
+      <p>
+        {' Posted by '}
```
Comment (Viktor-Kost) [inline src/components/PostInfo/PostInfo.jsx:10]: `span`

File: src/components/PostInfo/PostInfo.jsx [hunk 5]
```diff
@@ -1,3 +1,22 @@
...
+        <UserInfo {...user} />
```
Comment (Viktor-Kost) [inline src/components/PostInfo/PostInfo.jsx:11]: user can be undefined so add a condition to show this component only when user is exist

File: src/components/PostList/PostList.jsx [hunk 6]
```diff
@@ -1,3 +1,9 @@
-export const PostList = () => (
-  <>Put the list here</>
+import { PostInfo } from '../PostInfo';
+
+export const PostList = ({ posts }) => (
+  <div className="PostList">
+    {posts.map(post => (
+      <PostInfo key={post.id} {...post} />
```
Comment (Viktor-Kost) [inline src/components/PostList/PostList.jsx:6]: I recommend you to pass post as one instance 
```suggestion
      <PostInfo key={post.id} post={post} />
```

File: src/components/PostInfo/PostInfo.jsx [hunk 7]
```diff
@@ -1,3 +1,22 @@
-export const PostInfo = () => (
-  <>Put the post here</>
+import './PostInfo.scss';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
+
+export const PostInfo = ({ title, user, body, comments }) => (
```
Comment (Viktor-Kost) [inline src/components/PostInfo/PostInfo.jsx:5]: If you pass `post` as one instance it'd be better to work with types and make better props structure
```suggestion
export const PostInfo = ({ post}) => {
  const { title, user, body, comments } = post;
  
  return (
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

**Pull request closed without merging**